network-area-diagram/src/main/java/com/powsybl/nad/layout/AbstractLayout.java

```diff
@@ -76,9 +88,17 @@ protected void busNodesLayout(Graph graph) {
     }
 
     protected void fixedTextNodeLayout(Pair<VoltageLevelNode, TextNode> nodes, LayoutParameters layoutParameters) {
-        Point fixedShift = layoutParameters.getTextNodeFixedShift();
-        Point textPos = nodes.getFirst().getPosition().shift(fixedShift.getX(), fixedShift.getY());
+        String textNodeId = nodes.getSecond().getDiagramId();
+        Point textShift = textNodesWithFixedPosition.containsKey(textNodeId) ?
```

Comment:
you're calling twice containsKey and twice get on the map for each text node, you'd rather make a single call  `textNodesWithFixedPosition.get(voltageLevelId)` to be 4 times faster.